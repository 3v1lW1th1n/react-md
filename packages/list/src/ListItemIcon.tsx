/* eslint-disable react/prop-types, react/destructuring-assignment */
import React, { FC } from "react";
import cn from "classnames";
import { bem } from "@react-md/theme";
import { TextIconSpacing, TextIconSpacingProps } from "@react-md/icon";

export type ListItemIconPosition = "top" | "middle" | "bottom";

export interface ListItemIconProps extends TextIconSpacingProps {
  before: boolean;
  avatar: boolean;
  media: boolean;
  mediaLarge: boolean;
  position: ListItemIconPosition;
}

const base = bem("rmd-list-item");

const ListItemIcon: FC<ListItemIconProps> = ({
  className,
  avatar,
  before,
  children,
  media,
  mediaLarge,
  position,
  ...props
}) => (
  <TextIconSpacing
    {...props}
    forceIconWrap={props.forceIconWrap || media}
    className={cn(
      base("icon", {
        [position]: position !== "middle",
        avatar,
        media: media || mediaLarge,
        "media-large": mediaLarge,
      }),
      className
    )}
    iconAfter={!before}
  >
    {children}
  </TextIconSpacing>
);

export default ListItemIcon;
