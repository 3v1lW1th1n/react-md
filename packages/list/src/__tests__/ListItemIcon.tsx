import React from "react";
import { render } from "@testing-library/react";

import ListItemIcon, { ListItemIconProps } from "../ListItemIcon";

describe("ListItemIcon", () => {
  it("should render correctly based on the addon type", () => {
    const props: ListItemIconProps = {
      children: "Content",
      position: "middle",
      before: false,
      avatar: false,
      media: false,
      mediaLarge: false,
    };

    const { container, rerender } = render(<ListItemIcon {...props} />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} before />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} avatar />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} avatar before />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} media />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} media before />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} mediaLarge />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} mediaLarge before />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} position="top" />);
    expect(container).toMatchSnapshot();

    rerender(<ListItemIcon {...props} position="bottom" />);
    expect(container).toMatchSnapshot();
  });
});
