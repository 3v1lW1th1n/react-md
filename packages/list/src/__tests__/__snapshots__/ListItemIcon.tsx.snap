// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemIcon should render correctly based on the addon type 1`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 2`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 3`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 4`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 5`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 6`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 7`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 8`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 9`] = `
<div>
  Content
</div>
`;

exports[`ListItemIcon should render correctly based on the addon type 10`] = `
<div>
  Content
</div>
`;
